095-deprecate-OPA-authorization.md

```diff
@@ -0,0 +1,87 @@
+
+This proposal suggests to deprecate and later remove the Open Policy Agent (OPA) authorization (`type: opa`).
+
+## Current situation
+
+Strimzi currently support [Open Policy Agent authorization plugin](https://github.com/StyraInc/opa-kafka-plugin) for Kafka brokers.
+The plugin is bundled as part of our Apache Kafka container images.
+Users can configure it in the `Kafka` custom resource using the `type: opa` authorization.
+For example:
+
+```yaml
+    authorization:
+      type: opa
+      url: http://opa:8181/v1/data/kafka/crd/authz/allow
+      expireAfterMs: 60000
+      superUsers:
+        - my-super-user
+```
+
+## Motivation
+
+Supporting the `type: opa` authorization and bundling the plugin in our images is not for free:
+* We need to maintain the code in the Cluster Operator
+* With every new Kafka release, we need to make sure the dependencies are aligned between Kafka, other plugins and the OPA Authorizer plugin
+* We need to maintain the system tests and make sure we use an reasonably up-to-date OPA version
+
+While there seem to be some users using the `type: opa` authorization, it does not seem to be widely used.
+For the users using the OPA authorizer, there is also a possible workaround.
+They can continue using the OPA authorizer plugin even after we remove the dedicated support for it by following these steps:
+1. Add the OPA authorizer plugin to the Kafka container image
+2. Use the `type: custom` authorization to configure the OPA authorizer.
+   The OPA authorizer class will be specified as part of the `authorization` section.
+   The additional options can be specified in the `config` section.
+   For example:
+   ```yaml
+   # ...
+   kafka:
+     # ...
+     authorization:
+       type: custom
+       authorizerClass: org.openpolicyagent.kafka.OpaAuthorizer
+       superUsers:
+         - my-super-user
+    config:
+      opa.authorizer.url: http://opa:8181/v1/data/kafka/crd/authz/allow
+      opa.authorizer.cache.expire.after.seconds: 60
+    # ...
+   ```
+
+Given the available workaround and the maintenance effort, it seems reasonable to deprecate and remove the direct support for OPA authorizer from Strimzi.
+It also helps to make Strimzi project leaner and rely more on pluggability instead.
+
+## Proposal
+
+This proposal suggests to immediately within Strimzi 0.46:
+* Deprecate the `type: opa` authorization
+* Update the documentation to not use this deprecated field and use the `type: custom` authorization instead
+* Update the `CHANGELOG.md` file and documentation to inform users about this deprecation
+* Have warnings raised by the Cluster Operator when the `type: opa` authorizer is used
+
+While deprecated, we will still continue bundling the OPA authorizer plugin as part of Strimzi.
+
+As part of Strimzi `v1` CRD API, we will remove `type: opa` authorization from the `v1` API.
+
+Finally, in the first Strimzi version that drops the support for the `v1beta2` API, we will:
```

Comment:
The previous sentence mentions to remove `opa` type as part of `v1` CRD.
The above one mentions a Strimzi version which drops `v1beta2` API.
How are they related? I mean, there will be a Strimzi version supporting both `v1` and `v1beta2` where `opa` type is removed (because of `v1`) but the following things will be still in place like bundling plugin, because still `v1beta2`? Can you elaborate and maybe clarify in the proposal please?

Reply:
The `v1` will not support `type: opa` API anymore. But while `v1beta2` is still supported, the API will be there. So the support needs to remain in the code and in the container images.

Reply:
I tried to make it more clear in the text.

Reply:
Ignore my last comment, I was adding it but while you were updating the proposal. Your new text is fine with me ;-)